### PR TITLE
Bump kubevirt to 0.58

### PR DESCRIPTION
**Pull request by rhrazdil**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:

```release-note
NONE
```

### CODE REVIEW

File: hack/functest.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ echo "Running e2e test suite"
 export KUBECONFIG=$(./kubevirtci kubeconfig)
 export IMAGE_REPO=k8s.gcr.io
 export TENANT_CLUSTER_KUBERNETES_VERSION="v1.23.10"
-export NODE_VM_IMAGE_TEMPLATE=quay.io/capk/ubuntu-2004-container-disk:${TENANT_CLUSTER_KUBERNETES_VERSION}
```
Comment (davidvossel) [inline hack/functest.sh:18]: once CI passes, we need to remember to promote your image to the quay/capk repo and update this line before merging.

**Comment by rhrazdil** [comment]
The CI is failing on something that's not related to this PR, however, there is another issue it seems.
The [test](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/main/e2e/create-cluster_test.go#L756)  gets pass the VM restart, so that works fine, but fails at this [step](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/main/e2e/create-cluster_test.go#L841) (when running locally):
```
By("wait for a new VMI to be created")
```
For some reason, the VM is stopped, so the VMI is never re-created. I'm not sure what causes it. I suspected the drain, but I'm failing to reproduce it locally.
@davidvossel

**Comment by rhrazdil** [comment]
I found the issue why the test was failing for me locally. For some reason, the test finalizer removed at this [step](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/main/e2e/create-cluster_test.go#L839) is not actually removed from the VMI.

**Comment by davidvossel** [comment]
ci is unfortuantely broken due a switch to podman that occurred recently. @nunnatsa is addressing these issues in pr https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/207

**Comment by rhrazdil** [comment]
There's apparently another flake when creating a client, but the CI has passed with the updated ubuntu image
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubernetes-sigs_cluster-api-provider-kubevirt/206/pull-kubernetes-sigs-cluster-api-provider-kubevirt-e2e/1592517095147442176

**Review by davidvossel — COMMENTED** [review]
/approve
/hold
 waiting for update with image in capk repo before merging.

**Comment by rhrazdil** [comment]
Updated the ubuntu image

**Comment by rhrazdil** [comment]
The updated ubuntu image was merged. This should be safe to merge now, wdyt @davidvossel ?

**Pull request merged**